src/SAWScript/MGU.hs

```diff
@@ -1276,9 +1330,30 @@ evalTIWithEnv env tenv m =
 --
 -- The third is a current position, and the fourth is the
 -- context/monad type associated with the execution.
-checkStmt :: Map LName Schema -> Map Name Type -> Pos -> Context -> Stmt -> Either [(Pos, String)] Stmt
+--
+-- XXX: we shouldn't need a position here.
+-- The position is used for the following things:
```

Comment:
With a the way this is written currently, the `XXX` comments (which I presume are meant to be internal-only) will leak out into the Haddocks. You can avoid this by adding a newline between the Haddock comments and the `XXX` comments:

```suggestion

-- XXX: we shouldn't need a position here.
-- The position is used for the following things:
```

Reply:
Since the Haddocks for this code are meant to be an internals reference, that doesn't seem like the end of the world, but I agree it's untidy, so I've fixed it as you suggested in be94c5fc1511f5aebdc2ee25d18cf5bbff2a86dd